### DISCUSSION TITLE
Using Cache::flexible in with redis

### DISCUSSION THREAD

**Discussion by MohamedAbdel-wahed** [laravel/framework#52858] [Q&A] [answered]
Cache::flexible is supposed to get the stale data from the cache and return it to the user while recalculating and setting the cache again in the background.

Now with redis:
If i give an expiry time (ttl) for redis cache, Redis will automatically delete the cache after the specified ttl ends

So my question here
How would laravel get the stale data in this case if the data was already deleted automatically by redis ?

Thanks for all 💥

**Comment by henzeb** [comment]
Well, reading the code under the hood, it doesn't give a timestamp to redis. As far as redis, or any other driver is concerned, it is a cache item that is to stay forever. there is a second cache item that contains the ttl. if that is reached, it will renew the data. depending on the lock configuration, which in my humble opinion is quite complex. I don't understand why or what that is doing.

but bottom line is, it doesn't rely on redis ttl at all.

**Reply by MohamedAbdel-wahed** [reply]
Tbh i am still confused

Did you mean that the redis cache will stay forever while laravel internally handles the ttl ?

If so, i want to say that i observed this myself in redis-insight. I noticed that redis cache is deleted automatically from redis itself.

**Reply by henzeb** [reply, marked as answer]
ah, looking again at the code, this time rearranging it so I can actually read this, it is indeed putting in a ttl. 

ttl is an array here, so you have to set the second farther away into the future thant the first one. then it might work. This is  what is called the grace period. Between those two periods the old cache is still returned while the request for creating new data is being deferred so that next request can have the new data.

imho way to complex and un-laravel-like method imho.
